Es6 configuration

erikperkins opened the pull request:
Fixes issue #15.

Set up app to use ES6.

rmacklin commented:
The commits https://github.com/AppFolioOnboarding/perkins_eidolon/commit/410febd449bac2f947f654ca5fcc8f5063da3134 and https://github.com/AppFolioOnboarding/perkins_eidolon/commit/f24bf470a92addb30e4f9b33672339a9cc885e69 are a bit intertwined. The first one includes "infrastructure changes" (as noted in the commit message) _as well as_ most of the work for adding the page-specific JS to the images index. It would be better to cleanly separate the two.

rmacklin commented on Gemfile:
```diff
@@ -35,6 +35,12 @@ gem 'sdoc', '~> 0.4.0', group: :doc
 gem 'puma'
 
+# Gem to handle image tags
+gem 'acts-as-taggable-on'
```
This change is unrelated to ES6 configuration. Did you mean to add this gem on another branch?

erikperkins replied:
Addressed in bff3498

rmacklin replied:
:+1: you can squash that change.

rmacklin commented on app.json:
```diff
@@ -4,11 +4,20 @@
   "addons": [
-
+    "heroku-postgresql:hobby-dev",
+    "sendgrid:starter"
```
Where did you get this from?

erikperkins replied:
During the initial application setup, my app.json file was generated empty. I stole this line from the app.json file in the vanilla app.

rmacklin replied:
I see. It'd be best not to copy things blindy :smile: - the goal is to understand every line of your changeset in a PR. Sendgrid is an email service, which we use in the Share Image story, but it's not necessary for this story, so we should remove it for now.

rmacklin commented on app/assets/javascripts/application.js:
```diff
@@ -13,4 +13,4 @@
 //= require jquery_ujs
 //= require turbolinks
-//= require_tree .
+// require_tree .
```
This line should just be removed altogether. It's a bit confusing to leave it like this.

erikperkins replied:
Addressed in 0743f5d

rmacklin replied:
:+1: you can squash that change.

rmacklin commented on config/initializers/assets.rb:
```diff
@@ -9,3 +9,8 @@
 # Rails.application.config.assets.precompile += %w( search.js )
+
+font = ['*.eot', '*.swf', '*.ttf', '*.woff', '*.woff2', '*.otf', '*.svg']
+Rails.application.config.assets.precompile += font
```
These changes seem unrelated to ES6 configuration. Perhaps they belong on another branch. (Additionally, you could also get fonts from a CDN instead of worrying about precompiling them and serving them ourselves.)

erikperkins replied:
Addressed in bff3498

rmacklin replied:
:+1: you can squash that change.

rmacklin commented:
Thanks for making those changes. Let me know if you'd like to work together to squash the changes into atomic commits.

erikperkins commented:
I rebased the intertwined commits.

rmacklin commented:
Thanks. I left some more comments on https://github.com/AppFolioOnboarding/perkins_eidolon/commit/d628158411355f73425b8b47eef0a9fd5e2cc99d and https://github.com/AppFolioOnboarding/perkins_eidolon/commit/cb0af287f542e05ae51a7318ba7217c7ce5f59dd. I think one more rebase is in order to address those and then this'll be good to go.

erikperkins commented:
I addressed your comments on d628158411355f73425b8b47eef0a9fd5e2cc99d and cb0af287f542e05ae51a7318ba7217c7ce5f59dd, and split the commit for app.json into two as you suggested.

rmacklin commented:
Looks good, although it looks like the review app had a problem deploying.

erikperkins commented:
@rmacklin The review app looks like it deployed successfully. Is it ready to merge now?

rmacklin commented:
Yes, and then you should make sure it works on your staging application (and your production application if you then promote staging to production).
